octojobs/views/default.py

```diff
@@ -0,0 +1,109 @@
+"""Render views from configurations."""
+
+from pyramid.view import view_config
+from sqlalchemy import and_
+
+from pyramid.httpexceptions import HTTPFound
+
+from ..models import Job
+
+
+@view_config(route_name='home', renderer='../templates/home.jinja2')
+def home_view(request):
+    """On initial load, shows search bar. On query submit, loads results."""
+    if request.method == 'POST':
+
+        return post_request(request)
+
+    return {}
+
+
+def redirect_search(request, query):
+    """Function to redirect to result page."""
+    return HTTPFound(location=request.route_url('results', _query=query))
+
+
+def post_request(request):
+    """If post request on form, collect search terms."""
+    keyword = request.POST['searchbar']
+    location = request.POST['location']
+
+    search_terms = {}
+
+    if not location and keyword:
+        search_terms['search'] = keyword
+        return redirect_search(request, search_terms)
+
+    elif not keyword and location:
+        search_terms['location'] = location
+        return redirect_search(request, search_terms)
+
+    elif keyword and location:
+        search_terms['location'] = location
+        search_terms['search'] = keyword
+        return redirect_search(request, search_terms)
+
+    else:
+        return {'no_query': 'no result'}
+
+
+@view_config(route_name='results', renderer='../templates/results.jinja2')
+def result_view(request):
+    """Generate results view."""
+    if request.method == 'POST':
+
+        return post_request(request)
+
+    try:
+        searchterm = '%' + request.GET.get('search') + '%'
+    except TypeError:
+        searchterm = None
+
+    try:
+        location = '%' + request.GET.get('location') + '%'
+    except TypeError:
+        location = None
```

Comment:
This exception handling could be a helper function (since it does the same thing twice but just looks for a given key)